petstore/src/main/scala/io/finch/petstore/reader.scala

```diff
@@ -0,0 +1,36 @@
+package io.finch.petstore
+
+object reader{
+  import model._
+
+  /*
+  A request reader that reads user objects from the http request.
+  - A user is represented by url-encoded param "username"
+   */
+//  val userReader: RequestReader[User] =
+//    param("username") should beLongerThan(4) map {
+//      name => User(Id(), username, "unknown", "unknown", "unknown", "unknown", "unknown")
+//    }
+
+  /*
+  RequestReader for Category
+   */
+  val petCategoryReader: RequestReader[Category] = {
```

Comment:
You only need `RequestReader[A]` instances that correspond to `body` or `query` parameters in the [API](http://petstore.swagger.io/).